### PR TITLE
Add Bitbucket support

### LINKED ISSUE #512: Add Bitbucket support
I think this should be pretty simple once dependabot/dependabot-core#2917 and dependabot/dependabot-core#2919 are resolved.

### PR DESCRIPTION
**Pull request by iinuwa**
This adds support (and documentation) for Bitbucket. It is not ready to be merged yet; it is dependent on dependabot/dependabot-core#2918 and dependabot/dependabot-core#2923 being merged. The Gemfile currently points to a fork with the prerequisite changes, so it should run locally.

Should fix #512.

(The large diff in README.md is due to building off of #513.)

### CODE REVIEW

**Comment by Jiehong** [comment]
Both referenced PRs on core are now merged. @iinuwa : can this be updated to be mergeable?

**Comment by Jiehong** [comment]
Both referenced PRs on core are now merged. @iinuwa : can this be updated to be mergeable?

**Comment by iinuwa** [comment]
@Jiehong, sure. Are you able to merge #513?

**Comment by johnmccabe** [comment]
@iinuwa is it possible to set the `BITBUCKET_API_URL` envvar to something other than `https://api.bitbucket.org/2.0/`?

**Comment by iinuwa** [comment]
@johnmccabe, technically, yes. It could be set to some sort of proxy URL or something.

I initially added this in case the user was using Bitbucket Server; however, I have since learned that Bitbucket Server is deprecated and will never support the Bitbucket REST 2.0 API. So this won't be helpful for on-premise installations.

**Comment by JaredAAT** [comment]
Hey there, looking to be able to add dependabot access to our bitbucket account.  Is this and #513 the last PR's to be merged before we can add dependabot?

**Comment by iinuwa** [comment]
@JaredAAT, yeah, I think so. If you want, you can use my fork; I just fixed the conflicts.

**Comment by iinuwa** [comment]
@Jiehong, I believe I rebased of of master, but CI is failing now, and I don't think anything in this PR did that. Do you know why this is failing?

File: generic-update-script.rb [hunk 1]
```diff
@@ -177,6 +195,7 @@
     files: updated_files,
     credentials: credentials,
     assignees: [(ENV["PULL_REQUESTS_ASSIGNEE"] || ENV["GITLAB_ASSIGNEE_ID"])&.to_i],
+    author_details: { name: "Dependabot", email: "support@dependabot.com" },
```
Comment (feelepxyz) [inline generic-update-script.rb:198]:
```suggestion
    author_details: { name: "Dependabot", email: "no-reply@github.com" },
```

**Comment by feelepxyz** [comment]
Porting https://github.com/dependabot/dependabot-script/pull/580

**Comment by iinuwa** [comment]
Thanks @feelepxyz!
@JaredAAT, Bitbucket support should be in `main` now.

**Pull request closed without merging**